apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx

```diff
@@ -142,45 +156,62 @@ const Body = ({ data, onClose }: { data: Data; onClose?: () => void | undefined
 
   const onBroadcastSuccess = useCallback(
     (operation: Operation) => {
-      // If swap we save to swap history and keep open the drawer
-      if (swapId && toAccount && magnitudeAwareRate && sourceCurrency && targetCurrency) {
-        const newResult = {
-          operation,
-          swapId,
-        };
-        updateAccount({
-          result: newResult,
-          magnitudeAwareRate,
-        });
-        setResult({
-          swapId,
-          provider,
-          sourceCurrency,
-          targetCurrency,
-        });
+      // TODO: swapId && sellId will become quoteId
+
+      const isValidExchangeSDKSwapTx =
+        (ExchangeType.SWAP || ExchangeType.SWAP_NG) &&
+        swapId &&
+        toAccount &&
+        magnitudeAwareRate &&
+        sourceCurrency &&
+        targetCurrency;
+
+      const isValidExchangeSDKSellTx =
+        (data.exchangeType === ExchangeType.SELL || data.exchangeType === ExchangeType.SELL_NG) &&
+        sourceCurrency;
+
+      if (isValidExchangeSDKSellTx || isValidExchangeSDKSwapTx) {
+        if (isValidExchangeSDKSwapTx) {
+          const newResult = {
+            operation,
+            swapId,
+          };
+
+          updateAccount({
+            result: newResult,
+            magnitudeAwareRate,
+          });
+        }
+
+        const result = isValidExchangeSDKSwapTx
+          ? { swapId, provider, sourceCurrency, targetCurrency, mode: ExchangeModeEnum.Swap }
+          : { provider, mode: ExchangeModeEnum.Sell, sourceCurrency };
+
+        setResult(result);
```

Comment:
Plz simplify with a simple if swap do X else if sell do Y. There is nothing in common in here.

Reply:
We're checking 3 times isValidExchangeSDKSwapTx, we can only do it once with something like this and without Ifs inside ifs 
```
if (isValidExchangeSDKSwapTx) {
    const newResult = { operation, swapId };
    updateAccount({ result: newResult, magnitudeAwareRate });

    const result = {
        swapId,
        provider,
        sourceCurrency,
        targetCurrency,
        mode: ExchangeModeEnum.Swap,
    };
    setResult(result);
} else if (isValidExchangeSDKSellTx) {
    const result = {
        provider,
        sourceCurrency,
        mode: ExchangeModeEnum.Sell,
    };
    setResult(result);
}```

Reply:
It is valid to me

Reply:
I am not sure if it would be simpler send the complete "data" object here, as we could have different needs in the future.
```
      if (quoteId) {
        if (mode === SWAP {
          const newResult = {
            operation,
            swapId,
          };

          updateAccount({
            result: newResult,
            magnitudeAwareRate,
          });
        }
         setResult(data);
      }
       
```